### ISSUE TITLE
[breaking] Remove support for AMD and UMD formats

### ISSUE THREAD

**Issue by hildjj** [peggyjs/peggy#491]
Discuss.  Does anyone still use these?

**Comment by dasa** [comment]
I'm still using UMD

**Comment by hildjj** [comment]
@dasa do you forsee using UMD for the indefinite future, or is there some reason in the future you would stop?

**Comment by dasa** [comment]
I could refactor to use ESM if necessary.

**Comment by hildjj** [comment]
I'm considering this for the upcoming major release.  Now would be a good time for arguments to keep UMD and AMD support.

**Comment by cl4cnam** [comment]
I just saw this issue and I'm still using UMD.

**Comment by hildjj** [comment]
You generate them into two seperate .js files, right?
What JS runtime(s) or browser(s) are you using?

**Comment by cl4cnam** [comment]
Yes, into two separate files on https://peggyjs.org/online.html and both the parsers are loaded in the same HTML page by '<script src=...'.

**Comment by hildjj** [comment]
If you're only using the parser in the browser, then you likely don't need UMD.  Use `--format globals --export-var <NAME FOR THIS PARSER>` on the command line.  Alternately, use `--format es`, `<script type="module">`, and `import * as parser1 from parser1` when you're ready to move to modules.

**Comment by hildjj** [comment]
IIRC, the use case for UMD is you want to both use it in the browser and by `require` in node.js.

**Comment by cl4cnam** [comment]
Maybe, I've misunderstood. I thought I used UMD (I am a newbie to these concepts). I thought that removal of support for AMD and UMD will remove "Parser variable For CommonJS" and the "Download CJS parser" button. Is this the case or not?

**Comment by hildjj** [comment]
First, I strongly recommend not using the online version to generate your production code.  It doesn't have many options, it's always the latest version instead of a version you can pin, and it would be painful to make it repeatable.  `npm install -D peggy` and use the CLI.

Second, the `Download CJS parser` runs code that isn't particularly well-tested, because it only runs in the browser.  It uses `format: "commonjs"`, then prepends the generated source with "[parser-var] = ", which works because that variable leaks into the global scope.

Nothing on the online version generates UMD.

You'll get better output from using the CLI with `--format globals -e FOO`, and using `window.FOO.parse()`.

**Comment by cl4cnam** [comment]
I understand that it's much better with the command line but I don't want to install node.js on my PC. Does peggyjs work with Deno?

**Comment by hildjj** [comment]
Regardless, if on the online version you use a different parser-var for each of your parsers, you should be able to run them in the same page.  No UMD involved in your use case.

**Comment by cl4cnam** [comment]
Thank you very much 🙂. So "Parser variable For CommonJS" and the "Download CJS parser" button won't disappear 🙂. But I'm going to try with Deno, though. It will be much better than online version. And thank you very much for Peggy: it's a very good tool🙂.

**Comment by hildjj** [comment]
I've opened #597 to ensure that the CLI works with Deno.  It doesn't at the moment without some hacking.

**Comment by hildjj** [comment]
After implementing this change then thinking about it some more, I've decided that it's not important to make.  It's not that much code to maintain.
